lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -3359,6 +3359,104 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
         rewriter.replaceOp(binder.op, inputSequence);
         return success();
       });
+  patterns.onOp(
+      "SequenceMap", 17,
+      [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Value inputSequence, additionalInput;
+        Torch::ListType resultType;
+        if (binder.tensorListOperandAtIndex(inputSequence, 0) ||
+            binder.tensorListResultType(resultType))
+          return failure();
+        // get additional input and check if it's a sequence
+        bool hasAdditionalInput = true, isAdditionalList = true;
+        if (binder.tensorListOperandAtIndex(additionalInput, 1)) {
+          isAdditionalList = false;
+          if (binder.tensorOperandAtIndex(additionalInput, 1))
+            hasAdditionalInput = false;
+        }
+
+        Region *bodyRegion;
+        if (binder.getRegionAtIndex(bodyRegion, 0)) {
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "Failed getting Body Region");
+        }
+
+        // create a for-like primLoopOp
+        // with the length of sequence as max iter_num
+        Value len = rewriter.create<Torch::AtenLenTOp>(
+            binder.getLoc(), rewriter.getType<Torch::IntType>(), inputSequence);
+        auto cstTrue = rewriter.create<Torch::ConstantBoolOp>(
+            binder.getLoc(), rewriter.getBoolAttr(true));
+        mlir::ImplicitLocOpBuilder b(binder.getLoc(), rewriter);
+        auto loop = b.create<Torch::PrimLoopOp>(resultType, len, cstTrue,
+                                                inputSequence);
```

Comment:
I'm not sure, since the documentation might be a bit ambiguous here, but is it possible that the output sequence contains tensors of a different type than the first input sequence? For example, what if the payload is like "Reshape"? Or "Cast"? 

It might be more robust to initialize a completely blank input sequence, and just append to it instead of passing the first input list and modifying it iteratively.